x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/step_define_form.tsx

```diff
@@ -500,459 +163,204 @@ export const StepDefineForm: FC<Props> = React.memo(({ overrides = {}, onChange,
-                        if (isAdvancedSourceEditorEnabled && sourceConfigUpdated) {
-                          setAdvancedSourceEditorSwitchModalVisible(true);
-                          return;
-                        }
-
-                        toggleAdvancedSourceEditor();
-                      }}
-                      data-test-subj="transformAdvancedQueryEditorSwitch"
-                    />
-                    {isAdvancedSourceEditorSwitchModalVisible && (
-                      <SwitchModal
-                        onCancel={() => setAdvancedSourceEditorSwitchModalVisible(false)}
-                        onConfirm={() => {
-                          setAdvancedSourceEditorSwitchModalVisible(false);
-                          toggleAdvancedSourceEditor(true);
-                        }}
-                        type={'source'}
+                    {!disabledQuery && !isAdvancedSourceEditorEnabled && (
+                      <SourceSearchBar
+                        indexPattern={indexPattern}
+                        searchBar={stepDefineForm.searchBar}
                       />
                     )}
+                    {isAdvancedSourceEditorEnabled && <AdvancedSourceEditor {...stepDefineForm} />}
                   </EuiFlexItem>
-                  {isAdvancedSourceEditorEnabled && (
-                    <EuiButton
-                      size="s"
-                      fill
-                      onClick={applyAdvancedSourceEditorChanges}
-                      disabled={!isAdvancedSourceEditorApplyButtonEnabled}
-                    >
-                      {i18n.translate(
-                        'xpack.transform.stepDefineForm.advancedSourceEditorApplyButtonText',
-                        {
-                          defaultMessage: 'Apply changes',
-                        }
+                </>
+              )}
+
+              {/* Search options: Advanced Editor Switch / Copy to Clipboard / Advanced Editor Apply Button */}
+              <EuiFlexItem grow={false} style={{ width: advancedEditorsSidebarWidth }}>
+                <EuiSpacer size="s" />
+                <EuiFlexGroup gutterSize="xs" direction="column" justifyContent="spaceBetween">
+                  <EuiFlexItem grow={false}>
+                    <EuiFlexGroup alignItems="center" justifyContent="spaceBetween">
+                      {searchItems.savedSearch === undefined && (
+                        <EuiFlexItem grow={false}>
+                          <AdvancedQueryEditorSwitch {...stepDefineForm} />
+                        </EuiFlexItem>
                       )}
-                    </EuiButton>
+                      <EuiFlexItem grow={false}>
+                        <EuiCopy
+                          beforeMessage={copyToClipboardSourceDescription}
+                          textToCopy={copyToClipboardSource}
+                        >
+                          {(copy: () => void) => (
+                            <EuiButtonIcon
```

Comment:
Wonder if this 'copy to clipboard' icon should always be right-aligned. When you come in from a saved search, and the query bar isn't showing, it moves over to the left:

![image](https://user-images.githubusercontent.com/7405507/80701050-ac812280-8ad6-11ea-90f5-0a8a39bb74d3.png)

Reply:
Tweaked the layout for saved searches in: 9862b97

![image](https://user-images.githubusercontent.com/230104/80959972-2fbcb400-8e08-11ea-80dc-be88e5a4c3ab.png)